session-3-dom-api/README.md

```diff
@@ -0,0 +1,347 @@
+# Hackschool Session 3: Webpage Interaction with the DOM
+
+**Date**: October 28, 2020
+
+**Location**: Zoom
+
+**Teacher**: Galen Wong ([@GalenWong](https://github.com/GalenWong))
+
+<!-- omit in toc -->
+## Resources
+
+- [Slides](https://docs.google.com/presentation/d/1UZ6EJ3UNhUvtJgBLDvHp0iAZ_1ReLZrjov1a5ujeiT8/edit?usp=sharing)
+- [ACM Membership Attendance Portal](http://members.uclaacm.com/login)
+- 
+<!-- omit in toc -->
+## What we'll be learning today
+
+
++ [Linking JavaScript to HTML](#linking-javascript-to-html)
++ [What is the DOM?](#what-is-the-dom)
+  + [Accessing individual elements](#accessing-individual-elements)
+  + [Manipulating the elements](#manipulating-the-elements)
+  + [A little detour: how are colors represented in a computer?](#a-little-detour-how-are-colors-represented-in-a-computer)
+  + [Scheduling for things to happen](#scheduling-for-things-to-happen)
+  + [Listening and reacting to events](#listening-and-reacting-to-events)
++ [Summary](#summary)
++ [⚡️ Challenges](#️-challenges)
+
+
+Today we will base our workshop off some starter code in
+[demo-starter](/demo-starter). If you would like to follow along, [download
+the file](/demo-starter.zip)! It has a HTML and a CSS file. If you open the HTML file in the
+browser, it looks like this:
+
+<div align=center>
+    <img alt="rendered starter HTML" src="images/starter-html.png" width="500px">
+</div>
+
+
+## Linking JavaScript to HTML
+
+Last time, we had SO MUCH FUN coding in JavaScript in the console. But once
+you close the browser window, it is gone forever (like my ex. who am I
+kidding... I don't have one). So you would have to type it in the console
+again. Frustrating... no... actually we don't need to.
+
+Let's create a JavaScript file and name it `script.js`. Inside, we will put a
+line of simple JavaScript:
+
+```js
+console.log("If you see this in browser, that means our script is running.");
+```
+
+To use this JavaScript code file in our webpage, we need to link it to our 
+HTML. Here we introduce a new HTML tag called `<script>`:
+
+```html
+<body>
+  <div id="clock-face">0:00:01 AM</div> 
+  <div class="btn-container">
+    <button class="btn">Press</button>
+  </div>
+  <!-- We add the following line -->
+  <script src="script.js"></script>
+</body>
+```
+
+Now, if you open the HTML `index.html` in your browser and launch the
+console, you will see our `console.log` is executed. That means we have
+linked our JavaScript file to the HTML successfully.
+
+> 🚩 Checkpoint: Link the JavaScript file and see the `console.log` output in
+the console.
+
+## What is the DOM?
+
+DOM stands for **Document Object Model**. It is a programming model that
+represents documents as objects. In our case, our document is the HTML
+document. The object here refers to the JavaScript object that we learned
+last time.
+
+This means that we can store our HTML elements (each tag is an element) as a
+variable in JavaScript and manipulate it as an object.
```

Comment:
we can store HTML elements as variables in JavaScript and manipulate them as objects, or
we can store an HTML element as a variable in JavaScript and manipulate it as an object